Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -1098,97 +1080,104 @@ namespace OpenInfraPlatform {
 				) const throw(...)
 				{
 					if (ifcloop.isOfType<typename IfcEntityTypesT::IfcPolyLoop>()) 
-					{						
+					{	
 						convertIfcCartesianPointVectorSkipDuplicates(ifcloop.as<typename IfcEntityTypesT::IfcPolyLoop>()->Polygon, loop);
-
-						// If first and last point have same coordinates, remove last point
-						while (loop.size() > 2) {
-							carve::geom3d::Vector& first = loop.front();
-							carve::geom3d::Vector& last = loop.back();
-
-							if (abs(first.x - last.x) < 0.00000001) {
-								if (abs(first.y - last.y) < 0.00000001) {
-									if (abs(first.z - last.z) < 0.00000001) {
-										loop.pop_back();
-										continue;
-									}
-								}
-							}
-							break;
-						}
+						convertIfcPolyLoop(loop);

```

Comment:
`convertIfcCartesianPointVectorSkipDuplicates` should be called within `convertIfcPolyLoop`

Reply:
Resolved: 472d3b1